app/mobile-content/pages/shared/page-input/page-input.component.ts

```diff
@@ -0,0 +1,160 @@
+import { Component, Input, OnInit } from '@angular/core';
+import { MdDialog, MdDialogRef } from '@angular/material';
+
+import { MobilePage } from '../mobile-page.model';
+import { MobilePageService } from '../mobile-page.service';
+import { MediaService } from '../../../media/shared/media.service';
+import { Medium, mediaType } from '../../../media/shared/medium.model';
+import { SelectMediumDialogComponent } from '../../../media/select-medium-dialog/select-medium-dialog.component';
+import { SelectPageDialogComponent } from '../../select-page-dialog/select-page-dialog.component';
+import { Status } from '../../../shared/status.model';
+
+@Component({
+  moduleId: module.id,
+  selector: 'hip-page-input',
+  styleUrls: ['page-input.component.css'],
+  templateUrl: 'page-input.component.html'
+})
+export class PageInputComponent implements OnInit {
+  @Input() page: MobilePage;
+  audioTitle = '';
+  imageTitle = '';
+  infoPages = new Map<number, MobilePage>();
```

Comment:
The number in this case is the page's id. I use it later to easily retrieve any page I need via `infoPages.get(id)` because `additionalInformationPages` attribute stores only the ids of info pages. Using an array here would make things unnecessarily complicated because I would need to search the array for a specific id every time I need a page.